### PR TITLE
[SYCL][COMPAT] Migrate bug fixes & refactor of get_*version APIs

### PR DESCRIPTION
**Pull request by joeatodd**
This PR migrates changes to `get_version`, `get_major_version` and `get_minor_version`, adds new free function equivalents which accept a `sycl::device` argument, and adds relevant tests and documentation.

### CODE REVIEW

File: sycl/include/syclcompat/device.hpp [hunk 1]
```diff
@@ -59,6 +59,33 @@
 namespace syclcompat {
 
 namespace detail {
+static void get_version(const sycl::device &dev, int &major, int &minor) {
+  // Version string has the following format:
+  // a. OpenCL<space><major.minor><space><vendor-specific-information>
+  // b. <major.minor>
```
[the remark this answers is not in the document]
Comment (joeatodd) [reply sycl/include/syclcompat/device.hpp:66]: Addressed :+1:

File: sycl/include/syclcompat/device.hpp [hunk 2]
```diff
@@ -59,6 +59,33 @@
 namespace syclcompat {
 
 namespace detail {
+static void get_version(const sycl::device &dev, int &major, int &minor) {
```
[the remark this answers is not in the document]
Comment (joeatodd) [reply sycl/include/syclcompat/device.hpp]: I split out `syclcompat::detail::parse_version_string` and wrote some tests for it :+1:

[the remark this answers is not in the document]
Comment (Alcpz) [reply sycl/include/syclcompat/device.hpp]: This looks great, thanks!

**Review by Alcpz — APPROVED** [review]
Thanks for the PR!

**Comment by joeatodd** [comment]
@intel/llvm-gatekeepers This is ready to merge :pray:

**Pull request merged**